create statistic endpoint + service methods to get popular competitions

Kornienko11004 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added methods to retrieve popular competitive events from cache and database.
  - Introduced a new API endpoint to fetch a list of competitive events with filtering and result limits.
  - Made the `GetById` endpoint accessible anonymously in the competitive event controller.
- **Tests**
  - Updated test setup to support future testing of competitive event retrieval features.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
create endpoint + service methods to get popular competitions

Commit:
fix tests

Commit:
add attribute AllowAnonimous instead duplicating GetById for public usage

DmyMi commented on OutOfSchool/OutOfSchool.WebApi/Controllers/V1/CompetitiveEventController.cs:
```diff
@@ -30,6 +30,7 @@ public CompetitiveEventController(ICompetitiveEventService service,/* IUserServi
     /// <param name="id">CompetitiveEvent id.</param>
     /// <returns>CompetitiveEvent.</returns>
+    [AllowAnonymous]
     [HasPermission(Permissions.CompetitiveEventRead)]
```
I agree with rabbit - if there is allwo anonymous there should not be any permissions, because they will conflict

Kornienko11004 replied:
remove HasPermission attribute

Kornienko11004 committed:
Update CompetitiveEventController.cs

Remove unnecessary attribute which allows access to this endpoint only by authorized users
